src/darwin/Framework/CHIPTests/MTROTAProviderTests.m

```diff
@@ -634,7 +664,67 @@ - (XCTestExpectation *)announceProviderToDevice:(MTRDevice *)device
+
+    dispatch_queue_t queue = dispatch_get_main_queue();
+    XCTestExpectation * updateDefaultOTAProviderExpectation =
+        [self expectationWithDescription:@"UpdateDefaultOTAProvider verified"];
+
+    [sController
+            getBaseDevice:kDeviceId1
+                    queue:queue
+        completionHandler:^(MTRBaseDevice * _Nullable device, NSError * _Nullable error) {
+            XCTAssertEqual(error.code, 0);
+            XCTAssertNotNil(device);
+
+            MTRBaseClusterOtaSoftwareUpdateRequestor * cluster =
+                [[MTRBaseClusterOtaSoftwareUpdateRequestor alloc] initWithDevice:device endpointID:@(0) queue:queue];
+            NSArray * value = [NSArray arrayWithObject:providerLocation];
+            [cluster
+                writeAttributeDefaultOTAProvidersWithValue:value
+                                                completion:^(NSError * _Nullable error) {
+                                                    XCTAssertNil(error);
+                                                    MTRReadParams * params = [[MTRReadParams alloc] init];
+                                                    params.filterByFabric = true;
+                                                    [cluster
+                                                        readAttributeDefaultOTAProvidersWithParams:params
+                                                                                        completion:^(NSArray * _Nullable readValue,
+                                                                                            NSError * _Nullable err) {
+                                                                                            XCTAssertEqual(error.code, 0);
+                                                                                            XCTAssertNotNil(readValue);
+                                                                                            XCTAssertTrue(
+                                                                                                [self AreEqual:[readValue
+                                                                                                                   objectAtIndex:0]
+                                                                                                    providerLocation:
+                                                                                                        providerLocation]);
+                                                                                            [updateDefaultOTAProviderExpectation
+                                                                                                fulfill];
+                                                                                        }];
+                                                }];
+        }];
+
+    [self waitForExpectations:@[ updateDefaultOTAProviderExpectation ] timeout:kPairingTimeoutInSeconds];
```

Comment:
This should not be using kPairingTimeoutInSeconds; there is no pairing involved here.

Reply:
agreed. its a typo here. i tested and it works with just ktimeout. my bad

Reply:
N/A as I removed this for now